---
float64 fx
float64 fy
float64 fz
float64 tx
float64 ty
float64 tz